app/services/users/delete_activity.rb

```diff
@@ -0,0 +1,17 @@
+module Users
+  module DeleteActivity
+    module_function
+
+    def call(user)
```

Comment:
Would it be possible to add a spec for that? One where we run this and then validate that every related record is deleted. Probably have to be a little hacky and not sure how much of a PITA that would be but might be worth it. I could see this being forgotten pretty easily.

Reply:
I agree it would be nice to have a test for this though we'll need to prepare a lot of data for it. I will keep it as an opportunity for a further improvement.